src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -600,5 +613,89 @@ private static bool IsValidContributer(DataType dataType, ClaimsPrincipal user)
 
             return false;
         }
+
+        /// <summary>
+        /// Validated that the request 
+        /// </summary>
+        /// <param name="dataType"></param>
+        /// <param name="errorMessage"></param>
+        /// <returns></returns>
+        private bool CompliesWithDataRestrictions(DataType dataType, out string errorMessage)
+        {
+            errorMessage = string.Empty;
+
+            if (!Request.Headers.ContainsKey("Content-Disposition"))
+            {
+                errorMessage = "Conent-Disposition header containing 'filename' must be included in request.";
+                return false;
+            }
+
+            Request.Headers.TryGetValue("Content-Disposition", out StringValues headerValues);
+            string filename = GetFilenameFromHeader(headerValues.ToString());
+
+            if (string.IsNullOrEmpty(filename))
+            {
+                errorMessage = "Content-Disposition header must contain 'filename'.";
+                return false;
+            }
+
+            string[] splitFilename = filename.Split('.');
+
+            if (splitFilename.Count() != 2)
+            {
+                errorMessage = "Invalid format for filename. Expected format is {filename}.{filetype}.";
```

Comment:
This might be too strict. What if we instead did our best to find the file extension and if we don't find any or are unable to map the result to a mimeType, then we give an error?

I would like us to support multiple '.'. File extension should then be the last element. Some files can start with '.'. In that case we would probably be unable to find a mimeType. Thoughts?

Reply:
Reasonable.. I will re-implement the logic for finding filetype

Reply:
I've tried to make the filename format validation less strict now. Take a look a see what you think